### ISSUE TITLE
BLOCKER: Gradle maven-publish plugin doesn't work with latest spring-boot-gradle-plugin

### ISSUE THREAD

**Issue by asarkar** [gradle/gradle#1969]
With Spring Boot 1.5.3, all our projects are failing with the following error message:

> Error:Cannot configure the 'publishing' extension after it has been accessed.

There is some discussion about this error [here](https://discuss.gradle.org/t/apply-maven-publish-plugin-from-init-script/2460) and on [SO](http://stackoverflow.com/questions/36079404/errorcannot-configure-the-publishing-extension-after-it-has-been-accessed).

This is quite a frustrating issue that several people have reported previously; apparently, the Gradle team is "incubating" the publishing plugin for the last 3-4 years. Isn't it high time to turn this into a fully-supported plugin or drop it altogether?

Here's a Gradle file that can be used to reproduce this issue.
```
buildscript {
	ext {
		springBootVersion = '1.5.3.RELEASE'
	}
	repositories {
		jcenter()
	}
	dependencies {
		classpath(
				"org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}"
		)
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'

version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


ext {
	springCloudVersion = 'Dalston.RELEASE'
	artifactoryUrl = 'http://doesnotexist'
}

dependencies {
	testCompile('org.springframework.boot:spring-boot-starter-test')
}

dependencyManagement {
	imports {
		mavenBom "org.springframework.cloud:spring-cloud-dependencies:${springCloudVersion}"
	}
}

apply plugin: 'maven-publish'

project.publishing.publications {
	mavenJava(MavenPublication) {
		from components.java
	}
}

def isSnapshot = project.version.toString().toLowerCase().contains('snapshot')
logger.info("Publishing project version: ${project.version}")
def artifactoryUrl = "${artifactoryUrl}/${isSnapshot ? 'libs-snapshot-local' : 'libs-release-local'}"

project.publishing.repositories {
	maven {
		credentials {
			username "whatever"
			password "shhh"
		}
		url "${artifactoryUrl}"
	}
}
```

**Comment by eriwen** [comment]
@asarkar I hear you about the time-to-resolution of publishing issues. There's a lot of work to finish "incubating" features in Gradle, and you'll be happy to know this is one of them we'll be picking up in the next few months after we finish up a few other major things. 

Thanks for the reproducible example. I'm sure we'll put it to good use when verifying our improvements.

**Comment by asarkar** [comment]
Related to https://github.com/gradle/gradle/issues/1373

**Comment by oehme** [comment]
Fixed by #4997
